### ISSUE TITLE
add_trend

### ISSUE THREAD

**Issue by seabbs** [reconverse/i2extras#12]
**Is your feature request related to a problem? Please describe.**

Moving averages that are commonly used to process observational data prior to visualisation etc have several issues with the most notable being information loss and lag. The main drivers of the use of moving averages are day of the week effects and reporting noise. 

**Describe the solution you'd like**

In general, day of the week effects are much easier to account for than reporting effects. A simple model that includes a variable for each day of the week (but ideally using something more complex with fewer dofs like a sin cos sum) that is exposed as a feature like `show_trend` or `remove_day_of_week` would provide a nice alternative to a moving average. Time series decomposition methods could also be a nice backend. 

Reporting noise is harder to adjust for but allowing users to fit a simple negative binomial model etc would probably go along way to helping with this. An example could be `adjust_for_reporting(day_of_week = TRUE, model = "negbin")`. In the code this would fit the required model and then predict new values adjusted for reporting noise and the day of the week effect. 

I would in principle be happy to support the implementation of some of these features.

**Comment by tjtnew** [comment]
@seabbs This would be great. I need to change fit_curve a little (and possibly rename) so it seems like the option to include some form of weekday effect would be beneficial here. Let me start a discussion for this. Will update issue with link when ready.

**Comment by seabbs** [comment]
Makes sense. If I am right `fit_curve` is also assuming a static growth rate at the moment so its not totally ideal to replace a moving average.

**Comment by tjtnew** [comment]
moving to https://github.com/reconverse/i2extras/discussions/14 but will leave this open for the time being whilst I get to grips with discussions!
